source/common/upstream/cluster_manager_impl.cc

```diff
@@ -408,12 +424,21 @@ void ClusterManagerImpl::postThreadLocalClusterUpdate(
   HostListsConstSharedPtr healthy_hosts_per_locality_copy(
       new std::vector<std::vector<HostSharedPtr>>(primary_cluster.healthyHostsPerLocality()));
 
-  tls_->runOnAllThreads([this, name, hosts_copy, healthy_hosts_copy, hosts_per_locality_copy,
-                         healthy_hosts_per_locality_copy, hosts_added, hosts_removed]() -> void {
-    ThreadLocalClusterManagerImpl::updateClusterMembership(
-        name, hosts_copy, healthy_hosts_copy, hosts_per_locality_copy,
-        healthy_hosts_per_locality_copy, hosts_added, hosts_removed, *tls_);
-  });
+  tls_->runOnAllThreads([
+    this, name = primary_cluster.info()->name(), hosts_copy, healthy_hosts_copy,
```

Comment:
C++14 voodoo, seems unrelated but I like it.

Reply:
Yeah I had done this for my actual change and did this one also, but then didn't end up needing it. I can revert this if you want, but it is a lot nicer.

Reply:
It's fine, I learnt something new about C++14 today.

Reply:
Yeah this is a great feature.